contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,270 @@
+
+        _accTokenPerShare = _accTokenPerShare.add(_tokenReward.div(lpSupply));
+        return user.amount.mul(_accTokenPerShare).div(PRECISION).sub(user.rewardDebt);
+    }
+
+    /**
+     * @notice Withdraw JOE from sJOE and harvest the rewards
+     * @param _amount The amount of JOE to withdraw
+     */
+    function withdraw(uint256 _amount) external {
+        UserInfo storage user = userInfo[msg.sender];
+        require(user.amount >= _amount, "StableJoeStaking: withdraw amount exceeds balance");
+
+        updatePool();
+        uint256 pending = user.amount.mul(accTokenPerShare).div(PRECISION).sub(user.rewardDebt);
+
+        user.amount = user.amount.sub(_amount);
+        user.rewardDebt = user.amount.mul(accTokenPerShare).div(PRECISION);
+
+        safeTokenTransfer(msg.sender, pending);
+        joe.safeTransfer(address(msg.sender), _amount);
+        emit Withdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Withdraw without caring about rewards. EMERGENCY ONLY
+     */
+    function emergencyWithdraw() external {
+        UserInfo storage user = userInfo[msg.sender];
+
+        uint256 _amount = user.amount;
+        user.amount = 0;
+        user.rewardDebt = 0;
+
+        joe.safeTransfer(address(msg.sender), _amount);
+        emit EmergencyWithdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date
+     * @dev Needs to be called before any deposit or withdrawal
+     */
+    function updatePool() public {
+        if (block.timestamp <= lastRewardTimestamp) {
+            return;
+        }
+
+        uint256 lpSupply = joe.balanceOf(address(this));
+        if (lpSupply == 0) {
+            lastRewardTimestamp = block.timestamp;
+            return;
+        }
+
+        uint256 multiplier;
+        uint256 tokenReward;
+        // update values to be accurate with today's buyback
+        if (block.timestamp > lastDailyFeeTimestamp + 1 days) {
+            // remaining time of the previous day
+            multiplier = 1 days - (lastRewardTimestamp % 1 days);
+
+            tokenReward = multiplier.mul(tokensPerSecScaled);
+
+            // now we update the values for the new day
+            lastRewardTimestamp = lastDailyFeeTimestamp + 1 days;
+            // get the current day at 00:00:00, _lastRewardTimestamp and _lastDailyFeeTimestamp
+            // can be different, if `updatePool` isn't called at least once per day.
+            lastDailyFeeTimestamp = (block.timestamp / 1 days) * 1 days;
+
+            // get today's buyback amount
+            uint256 accruedFee = rewardToken.balanceOf(address(this)).sub(lastTokenBalance);
+            lastTokenBalance = lastTokenBalance.add(accruedFee);
+
+            // distribute today's buyback over 1 day
+            tokensPerSecScaled = accruedFee.mul(PRECISION).div(1 days);
+        }
+
+        // in case `updatePool` isn't called in more than 2 days, we need to make sure user receives only one day
```

Comment:
In fact it's not called during one full day, so
if it's called twice, one time at day 0, 23:59:59 and the second time at day 2 00:00:01 (no call during 24:00:02), it will be true
but if it's called twice, one time at day 0, 00:00:01 and the second time at day 1 23:59:59 (no call during 47:59:58), it will be true

Updated comments to `In case `updatePool` isn't called at least once per day`